sdks/python/apache_beam/io/gcp/pubsub_test.py

```diff
@@ -43,25 +46,48 @@
 class TestReadStringsFromPubSub(unittest.TestCase):
   def test_expand_with_topic(self):
     p = TestPipeline()
-    pcoll = p | ReadStringsFromPubSub('projects/fakeprj/topics/a_topic',
-                                      None, 'a_label')
-    # Ensure that the output type is str
+    p.options.view_as(StandardOptions).streaming = True
+    pcoll = (p
+             | ReadStringsFromPubSub('projects/fakeprj/topics/a_topic',
+                                     None, 'a_label')
+             | Map(lambda x: x))
+    # Ensure that the output type is str.
     self.assertEqual(unicode, pcoll.element_type)
 
+    # Apply the necessary PTransformOverrides.
+    overrides = _get_transform_overrides(p.options)
+    p.replace_all(overrides)
+
+    # Note that the direct output of ReadStringsFromPubSub will be replaced
```

Comment:
<!--thread_id:cc_165758525_t; commit:27878b6ef3c912e799ab3f3fd12da5a3538b354a; resolved:1-->
<!--section:context-quote-->
> **robertwb** wrote:
> These tests seem rather brittle. Is there a better way to test this transform application than grabbing the internal source and verifying a couple of properties on it. https://beam.apache.org/documentation/pipelines/test-your-pipeline/

<!--section:body-->
Thanks, and I agree, but the issue is that there really isn't anything to test.

The test here isn't a test of the transform; rather, it was (and still is) testing the behavior of replacing the transform with the correct DirectRunner replacement.  The transform itself is just a wrapper with runner-specific overrides.

Reply:
Sorry, I meant https://beam.apache.org/contribute/ptransform-style-guide/#testing-transform-construction-and-validation

If this is about the direct runner, we should put it into the direct runner tests. Best is if we could create a mock/in memory PubSub and make sure this works end-to-end (on any runner).

Reply:
But, as mentioned, fixing these existing tests should not block this PR. Please file a JIRA.

Reply:
Thanks, filed https://issues.apache.org/jira/browse/BEAM-3619.